lib/auth/auth_with_roles.go

```diff
@@ -3512,6 +3512,9 @@ func checkRoleFeatureSupport(role types.Role) error {
 	case modules.GetModules().BuildType() != modules.BuildEnterprise && len(allowReq.SearchAsRoles) != 0:
 		return trace.AccessDenied(
 			"role field allow.search_as_roles is only available in enterprise subscriptions")
+	case !features.DeviceTrust && options.DeviceTrustMode == constants.DeviceTrustModeRequired:
```

Comment:
Won't this break existing customers using device trust today?

Reply:
Not for enterprise customers since the teleport.e PR (https://github.com/gravitational/teleport.e/pull/1106) always sets `features.DeviceTrust` to `true`.

But for OSS users, if they happen to have `device_trust.mode=required` in a role, this change would prevent the auth server to start (since `checkRoleFeatureSupport` is called during startup).

We can change this to only return an error for enterprise builds, while just logging a warn for OSS. What do you think?

Reply:
Device trust is already enterprise only, so I'm not sure that's necessary.

/cc @codingllama

Reply:
No objections against the dedicated feature flag, but I think we need to do this in 3 steps: add the flag to OSS, flip it in Ent, then add the validations we see here. If we don't do that you are bound to break something, at least temporarily.

Today use the build type to decide whether device trust is available, so this code needs to change too: https://github.com/gravitational/teleport/blob/f83aa6238baddee6f5e3201db34d2ff038555f78/lib/devicetrust/config/config.go#L25-L31

You'll likely have to tweak a bunch of my e/ tests as well, give me a ping and I can help with some pointers.